std/math.d

```diff
@@ -249,8 +249,504 @@ enum RealFormat
     ieeeQuadruple,
 }
 
-// Constants used for extracting the components of the representation.
-// They supplement the built-in floating point properties.
+/* Low-level access to the individual parts of a floating-point value.
+
+   TODO: Make this publically available through std.bitmanip once the API
+   is stable and tested.
+*/
+union RealRep(N)
+if (isFloatingPoint!N &&  is(N == Unqual!N))
+{
+pure: @safe: nothrow: @nogc:
+pragma(inline, true):
+
+    // The floating-point number who's bit fields will be exposed.
+    N num = N.init;
+    alias num this;
+    this(const N num) inout
+    {
+        this.num = num;
+    }
+
+    static if (N.mant_dig == 106 && N.max_exp == 0x400)
+    {
+        /* N has IBM's double + double format, which is actually two ieeeDouble
+        (see below) stuck together. Pretending otherwise is not helpful at this
+        level. */
+        enum format = RealFormat.ibmExtended;
+
+        struct
+        {
+            version(LittleEndian)
+            {
+                RealRep!double lo;
+                RealRep!double hi;
+            }
+            else
+            {
+                RealRep!double hi;
+                RealRep!double lo;
+            }
+        }
+    }
+    else // N has some IEEE-like format:
+    {
+// Bitwise layout
+        /+
+            /* The number of "excess" binary digits of precision found at the
+            least significant end of the mantissa. Those digits should normally
+            be zero. This field is included to support ieeeExtended53. */
+            enum int excessDig;
+
+            /* True if this format explicitly stores the leading one of the
+            mantissa for normalized values. This field is included to support
+            ieeeExtended and ieeeExtended53. */
+            enum bool storeWhole;
+
+            // The number of binary digits in the raw exponent.
+            enum int expDig;
+
+            /* An unsigned integral type which is large enough to store the
+            complete mantissa (including the leading one) without loss of
+            precision. */
+            alias Mant;
+        +/
+
+        static if (N.mant_dig == 11 && N.max_exp == 0x6)
```

Comment:
I used hexadecimal for `N.max_exp` because for all other formats, `N.max_exp == (1 << (expDig - 1))` and hexadecimal makes it easy to verify at a glance that the numbers are actually correct.